xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -179,7 +181,7 @@ func (b *cdsBalancer) handleClientConnUpdate(update *ccUpdate) {
 		b.handleErrorFromUpdate(err, true)
 		return
 	}
-	b.clusterHandler.updateRootCluster(update.clusterName)
+	b.clusterHandler.updateRootCluster(strings.TrimPrefix(update.clusterName, clusterPrefix))
```

Comment:
Why would it be better to strip in the cluster manager lb policy? To me, it makes the most sense to strip it here, as the addition of the cluster: prefix was for cluster type cds. Stripping it that early leads me to think there would be name collisions somewhere (either in router or in cluster manager itself). Wdyt @menghanl?

Reply:
Discussed in person, decided it would be best it to strip this in the name resolver, while constructing the Cluster Manager LB Config itself. (kept the cluster: prefix for the name of the cluster, but stripped it for the underlying cds policy)